### PR TITLE
shell: Fix forwarding binary data for child frames

### PR DESCRIPTION
**Pull request by petervo**
We stopped doing base64 when we removed Hixie76 with #5621. So now we need to forward binary data from child frames.

### CODE REVIEW

File: pkg/shell/base_index.js [hunk 1]
```diff
@@ -317,7 +317,16 @@ var phantom_checkpoint = phantom_checkpoint || function () { };
             var forward_command = false;
             var data = event.data;
             var child = event.source;
-            if (!child || typeof data !== "string")
+            if (!child)
+                return;
+
+            /* If it's binary data just send it */
```
Comment (stefwalter) [inline pkg/shell/base_index.js]: Can we put a TODO here: ```/* TODO: Once we start restricting what frames can talk to which hosts, we need to parse control messages here, and cross check channels */```

Comment (petervo) [inline pkg/shell/base_index.js]: Added the TODO, but do we need to allow binary control messages?

**Comment by petervo** [comment]
> Looks straightforward. Is this breaking anything visible that could be exposed in a test, or not being used by cockpit itself?

Added a test for the docker terminal which is how i found the bug.

**Comment by stefwalter** [comment]
This needs a backport.

File: pkg/shell/base_index.js (hunk 1 above)
Comment (stefwalter) [inline pkg/shell/base_index.js]: Good point. And thinkng about this more we may have to go with the solution you proposed of having multiple WebSockets.

**Comment by stefwalter** [comment]
@petervo Will you open a new pull request with a backport to the rhel-7.4 and rhel-7.3.5 branches?

**Pull request merged**